text/0000-spellcheck-global-attribute.md

```diff
@@ -0,0 +1,164 @@
+
+**Table A:** Static string value, `<textarea spellcheck="staticValue">` vs `<c-foo spellcheck="staticValue">`
+
+|Value\Rendered Element	|textarea	|c-foo	|textarea.spellcheck / c-foo.spellcheck	|
+|---	|---	|---	|---	|
+|`""`	|spellcheck	|spellcheck="false"	|true / false	|
+|`"false"` (case insensitive)	|spellcheck="false"	|spellcheck="true"	|false / true	|
+|`"any other string"`	|spellcheck="any other string"	|spellcheck="true"	|true / true	|
+
+**Table B:** Expression value, `<textarea spellcheck={expr}>` vs `<c-foo spellcheck={expr}>`
+
+|Value\Rendered Element	|textarea	|c-foo	|textarea.spellcheck / c-foo.spellcheck	|
+|---	|---	|---	|---	|
+|`undefined`	|not rendered (default state)	|spellcheck="false"	|**inherited **/ false	|
+|`null`	|not rendered (default state)	|spellcheck="false"	|**inherited **/ false	|
+|`0`	|spellcheck="0"	|spellcheck="false	|true / false	|
+|`false`	|spellcheck="false"	|spellcheck="false"	|false / false	|
+|`true`	|spellcheck="true"	|spellcheck="true"	|true / true	|
+|`""`	|spellcheck	|spellcheck="false"	|true / false	|
+|`"false"` (case insensitive)	|spellcheck="false"	|spellcheck="true"	|false / true	|
+|`"any other string"`	|spellcheck="any other string"	|spellcheck="true"	|true / true	|
+|object like	|spellcheck="val.toString()"	|spellcheck="true"	|true / true	|
+
+## Detailed design
+
+This RFC is attempting to solve this problem by changing the compiler to set the correct value into the `spellcheck` property when rendering the custom element, so it can reflect the value into the `spellcheck` attribute as `spellcheck="true"` or `spellcheck="false"`.
+
+### Static Values:
+
+In the case of static values, the solution is simple:
+
+> When compiling the template, we can check if the `spellcheck` attribute is being set to a string value, and normalize its value to boolean.
+
+With this solution Table A becomes:
+
+|Value\Rendered Element	|textarea	|c-foo	|textarea.spellcheck / c-foo.spellcheck	|
+|---	|---	|---	|---	|
+|`""`	|spellcheck	|spellcheck="true"	|true / true	|
+|`"false"` (case insensitive)	|spellcheck="false"	|spellcheck="false"	|false / false	|
+|`"any other string"`	|spellcheck="any other string"	|spellcheck="true"	|true / true	|
+
+### Dynamic Values (via template expressions):
+
+This case is more complicated because of all the invariants that it involves, but we can group them in 2 general cases that can be handled differently for custom elements:
+
+```html
+<template>
+  <c-foo spellcheck={expr}></c-foo> <!-- should reflect the attribute depending on the value of {expr} -->
+</template>
+```
+
+1. `expr == null` (`expr` result is `null` or `undefined`) This is the more complicated code-path, and the one that deserves more attention: 
+    1. Given that we always set the property, the attribute is always going to be set to `true` or `false` 
```

Comment:
>  Given that we always set the property

What if we were to change this invariant, to handle this case? Only set it the spellcheck property when it is not `null` or `undefined` ?

Reply:
then what will happen in such case that the expression changes from `true`/`false` to `null` or `undefined`?

Reply:
In this case we would invoke `removeAttribute`.

Reply:
no @pmdartus, remember that the main point here is that it MUST be a prop because we don't know if you have redefined that in your component via `@api spellcheck`, in which case you, as the author of the component, are in control.

Reply:
Yes, you are right.